### PR TITLE
Allow cross-compiling from Linux with MinGW

### PR DESCRIPTION
**Pull request by SpaceManiac**
I have no idea if it works, but it does build.

```sh
mkdir build
cd build
cmake ../byond-extools/ -DCMAKE_TOOLCHAIN_FILE=../byond-extools/cmake/i686-w64-mingw32.cmake
make
```

### CODE REVIEW

**Comment by ghost** [comment]
```
runtime error: Unable to load library byond-extools.dll
proc name: New (/world/New)
  source file: extools_testing.dm,74
  usr: null
  src: world
  call stack:
world: New()
```

```
steam@magrathea:/mnt/d/Code/C++/extools/byond-extools/build-linux$ file byond-extools.dll
byond-extools.dll: PE32 executable (DLL) (console) Intel 80386, for MS Windows
```

File: byond-extools/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,11 @@ add_library(byond-extools SHARED ${SRC_FILES})
 source_group(TREE ${SRC_DIR} FILES ${SRC_FILES})
 if (WIN32)
     set_property(TARGET byond-extools PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-    target_compile_options(byond-extools PRIVATE "/MP")
+    if (MSVC)
+        target_compile_options(byond-extools PRIVATE "/MP")
+    else()
+        target_link_libraries(byond-extools PRIVATE "ws2_32" "psapi")
```
Comment (ghost) [inline byond-extools/CMakeLists.txt]:
```suggestion
        target_link_libraries(byond-extools PRIVATE "ws2_32" "psapi" "-static-libgcc" "-static-libstdc++" "-static")
```

**Review by ghost — CHANGES_REQUESTED** [review]
You need to statically link libgcc and libstdc++ for it to work

File: byond-extools/CMakeLists.txt (hunk 1 above)
Comment (SpaceManiac) [reply byond-extools/CMakeLists.txt]: I don't love this but it can be revisited later

**Pull request merged**